### ISSUE TITLE
how to use sentecepiece tensorflow Op

### ISSUE THREAD

**Issue by kweonwooj** [google/sentencepiece#125]
I see you have added Tensorflow Op in your recent master.
Does this mean we can use SentencePiece Op in tensorflow graph?

I wish to build `libtensorflow_inference.so` file with SentencePieceOp integrated in, so that I can have 
a single Tensorflow Graph to make inferences.

Can you explain how we can utilize your recent master merged tensorflow op in this way?

Thanks in advance,

**Comment by taku910** [comment]
Yes, I've recently added tf ops, though it is still experimental. Hopefully we would like to make it available on tf.contrib.

As you mentioned, SentencePiece will be an op of Tensorflow, meaning that we can directly feed raw sentences with tf.placeholder to perform training and inference. The model deployment will be simplified.

You can try it out with "pip install tf_sentencepiece"   (only available for Linux).  
Here is the sample code.

https://colab.research.google.com/drive/1rQ0tgXmHv02sMO6VdTO0yYaTvc1Yv1yP?authuser=2#scrollTo=i_44FzFC0qwe

The model file (model proto) is an attribute of SentencePiece op.   SentencePiece model file is embedded into the TF graph so the model gets purely self-contained. 

Please let me know if you have any questions and comment.

**Comment by kweonwooj** [comment]
Thanks for the prompt response.
Great code and great implementation!

Which version of tensorflow do you think will Sentencepiece be available? (can't wait to see :))

**Comment by taku910** [comment]
I'm not sure when gets available. I will keep you posted.

**Comment by sooheon** [comment]
The op currently seems to only pad with UNK (idx 0), with no way to specify pad idx. 

Do you find that because Sentencepiece handles OOV well, symbols are not commonly encoded to UNK, and it is OK to use UNK as padding?

**Comment by taku910** [comment]
Right,  pad_id is not used in tf ops. We will fix it soon.

**Comment by jfhidakatsu** [comment]
I've tried using tf_sentencepiece, but whenever I try the line "import tf_sentencepiece as tfs", it gives the error:

'r: /home/ubuntu/.local/lib/python3.6/site-packages/tf_sentencepiece/_sentencepiece_processor_ops.so: undefined symbol: _ZN10tensorflow6StatusC1ENS_5error4CodeENS_11StringPieceE'

How can I import this library?

**Comment by taku910** [comment]
Please use the latest version. It supports tf 0.9 - 1.2

**Comment by taku910** [comment]
Added https://github.com/google/sentencepiece/blob/master/tensorflow/README.md

SentencePiece will not be integrated to tf.contrib because tf.contrib is going to be deprecated in TensorFlow 2.0.
Let me close this bug.

**Comment by tgsmith61591** [comment]
Sorry to comment on a closed thread... but has anyone had any luck building Tensorflow serving with the sentencepiece dependency?

**Comment by hbrylkowski** [comment]
@tgsmith61591 - did you manage to build it?

**Comment by tpacker** [comment]
Is there any documentation/examples on how to use SentencePiece as part of a trainable Tensorflow model, e.g. one based on a pretrained Universal Sentence Encoder lite?  I'm not finding anything online.
